### PR TITLE
[Merged by Bors] - mesh: store unapplied transactions without map

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

We won't be able to encode maps with ssz, and also if there will be a lot of unnapplied transactions we will have same problem as with tortoise.

## Changes
- rework unapplied transaction using leveldb batches and iteration

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: mesh/meshdb.go [hunk 1]
```diff
@@ -824,70 +825,43 @@ func (m *DB) removeFromUnappliedTxs(accepted []*types.Transaction) (grouped map[
...
+func (m *DB) removePending(account types.Address, txs []*types.Transaction) error {
+	var (
+		batch = m.unappliedTxs.NewBatch()
+		b     bytes.Buffer
+	)
+	for _, tx := range txs {
+		b.Write(account[:])
+		b.Write(tx.ID().Bytes())
+		if err := batch.Delete(b.Bytes()); err != nil {
+			return err
```
Comment (countvonzero) [inline mesh/meshdb.go:822]: nit: should this continue to delete as best effort?

**Review by countvonzero — APPROVED** [review]

File: mesh/meshdb.go (hunk 1 above)
Comment (dshulyak) [reply mesh/meshdb.go:822]: i think db writes doesn't produce temporary errors. if write failed something must be wrong with disk (or os) and the best thing we can do is to exit

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

File: mesh/meshdb.go [hunk 2]
```diff
@@ -787,29 +783,34 @@ func (m *DB) GetRewardsBySmesherID(smesherID types.NodeID) (rewards []types.Rewa
...
 
-	// TODO: instead of storing a list, use LevelDB's prefixed keys and then iterate all relevant keys
-	pending, err := m.getAccountPendingTxs(addr)
-	if err != nil {
-		return err
-	}
-	pending.Add(layer, accountTxs...)
-	if err := m.storeAccountPendingTxs(addr, pending); err != nil {
-		return err
+		b.Write(addr[:])
```
Comment (antonlerner) [inline mesh/meshdb.go]: for consistency use `addr.bytes()`

File: mesh/meshdb.go [hunk 3]
```diff
@@ -824,70 +825,43 @@ func (m *DB) removeFromUnappliedTxs(accepted []*types.Transaction) (grouped map[
...
-	if pending.IsEmpty() {
-		if err := m.unappliedTxs.Delete(account.Bytes()); err != nil {
-			return fmt.Errorf("failed to delete empty pending txs for account %v: %v", account.Short(), err)
+func (m *DB) removePending(account types.Address, txs []*types.Transaction) error {
+	var (
+		batch = m.unappliedTxs.NewBatch()
+		b     bytes.Buffer
+	)
+	for _, tx := range txs {
+		b.Write(account[:])
```
Comment (antonlerner) [inline mesh/meshdb.go]: `account.bytes()`

**Review by antonlerner — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**